### PR TITLE
Site tour: Use driver.js

### PR DESCRIPTION
**Pull request by trymebytes**
## Problem
For more flexibilities and out of the box features, we are trying out the driver.js library.

## Solution

Include the driver.js library to power the site tour.

## Testing Instructions
- Checkout this branch in your local GlotPress
- Use this sample filter
```
add_filter( 'gp_tour', function( ) {
	return array(
		'ui-intro' => [
			[
				'title' => 'UI Introduction Tour',
				'color' => '#3939c7',
			],
			[
				'selector' => '.strings .original',
				'html' => 'This is the original string that needs translation'
			],
			[
				'selector' => 'textarea.foreign-text',
				'html' => 'Enter translation here',
			],
			[
				'selector' => '.sidebar-tabs',
				'html' => 'See meta tabs here',
			],
			[
				'selector' => '.button.ok',
				'html' => 'Click here to save translation',
			]
		],
		
	);
} );
```
## Screenshots or screencast
using-driverjs

### CODE REVIEW

File: gp-includes/assets-loader.php [hunk 1]
```diff
@@ -17,6 +17,7 @@ function gp_register_default_styles() {
 	// Register our base style.
 	wp_register_style( 'gp-base', $url . '/style' . $suffix, array(), filemtime( $path . 'style' . $suffix ) );
 	wp_register_style( 'gp-jquery-webui-popover', $url . '/jquery.webui-popover' . $suffix, array( 'gp-base' ), filemtime( $path . 'jquery.webui-popover' . $suffix ) );
+	wp_register_style( 'driver-js', 'https://cdn.jsdelivr.net/npm/driver.js@1.0.1/dist/driver.css', array(), '1.0.1' );
```
Comment (akirk) [inline gp-includes/assets-loader.php]: I think we should be using a local copy, so can you add this file under `assets/css` to the PR and use that?

File: gp-includes/assets-loader.php [hunk 2]
```diff
@@ -37,6 +38,7 @@ function gp_register_default_scripts() {
 	wp_register_script( 'gp-translations-page', $url . '/translations-page' . $suffix, array( 'gp-editor' ), filemtime( $path . 'translations-page' . $suffix ) );
 	wp_register_script( 'gp-mass-create-sets-page', $url . '/mass-create-sets-page' . $suffix, array( 'gp-editor' ), filemtime( $path . 'mass-create-sets-page' . $suffix ) );
 	wp_register_script( 'gp-tour', $url . '/tour' . $suffix, array( 'jquery', 'gp-jquery-webui-popover' ), filemtime( $path . 'tour' . $suffix ), false );
+	wp_register_script( 'driver-js', 'https://cdn.jsdelivr.net/npm/driver.js@1.0.1/dist/driver.js.iife.js', array(), '1.0.1' );
```
Comment (akirk) [inline gp-includes/assets-loader.php]: I think we should be using a local copy, so can you add this file under `assets/js` to the PR and use that?

**Review by akirk — COMMENTED** [review]
Nice, I am surprised how easy it looks to swap it out!

File: gp-includes/assets-loader.php (hunk 2 above)
Comment (akirk) [reply gp-includes/assets-loader.php]: [The current version is 1.3.0](https://github.com/kamranahmedse/driver.js/releases/tag/1.3.0), maybe we can use that

**Comment by akirk** [comment]
It works well if I run it on the first string. But things no longer work if I start in the second one. Upon first sight it doesn't appear as if it supported a jQuery like `:visible` selector. Is there anything we can do about it?

Screenshot 2023-10-09 at 14 39 29

File: gp-includes/assets-loader.php [hunk 3]
```diff
@@ -17,6 +17,7 @@ function gp_register_default_styles() {
 	// Register our base style.
 	wp_register_style( 'gp-base', $url . '/style' . $suffix, array(), filemtime( $path . 'style' . $suffix ) );
 	wp_register_style( 'gp-jquery-webui-popover', $url . '/jquery.webui-popover' . $suffix, array( 'gp-base' ), filemtime( $path . 'jquery.webui-popover' . $suffix ) );
+	wp_register_style( 'driver-js', $url . '/driver-js' . $suffix, array(), '1.3.0' );
```
Comment (amieiro) [inline gp-includes/assets-loader.php]: Please, change the hard-coded version to a one based in the filemtime:

```suggestion
	wp_register_style( 'driver-js', $url . '/driver-js' . $suffix, array(), filemtime( $path . 'driver-js' . $suffix ) );
```

File: gp-includes/assets-loader.php [hunk 4]
```diff
@@ -36,14 +37,16 @@ function gp_register_default_scripts() {
 	wp_register_script( 'gp-glossary', $url . '/glossary' . $suffix, array( 'gp-editor' ), filemtime( $path . 'glossary' . $suffix ) );
 	wp_register_script( 'gp-translations-page', $url . '/translations-page' . $suffix, array( 'gp-editor' ), filemtime( $path . 'translations-page' . $suffix ) );
 	wp_register_script( 'gp-mass-create-sets-page', $url . '/mass-create-sets-page' . $suffix, array( 'gp-editor' ), filemtime( $path . 'mass-create-sets-page' . $suffix ) );
+	wp_register_script( 'gp-jquery-webui-popover', $url . '/jquery.webui-popover' . $suffix, array( 'jquery' ), filemtime( $path . 'jquery.webui-popover' . $suffix ), false );
 	wp_register_script( 'gp-tour', $url . '/tour' . $suffix, array( 'jquery', 'gp-jquery-webui-popover' ), filemtime( $path . 'tour' . $suffix ), false );
+	wp_register_script( 'driver-js', $url . '/driver-js' . $suffix, array(), '1.3.0' );
```
Comment (amieiro) [inline gp-includes/assets-loader.php]: Please, change the hard-coded version to a one based in the filemtime:

```suggestion
	wp_register_script( 'driver-js', $url . '/driver-js' . $suffix, array(), filemtime( $path . 'driver-js' . $suffix ) );

```

**Review by amieiro — DISMISSED** [review]

**Comment by trymebytes** [comment]
Converting this to a draft while I work on the patch to driver.js in a forked repository for easy referencing.

**Review by akirk — APPROVED** [review]
Works well!

**Pull request merged**